### PR TITLE
102 Drag Mode

### PR DESCRIPTION
**Pull request by DawnTheWitch**

### CODE REVIEW

File: src/DragMode.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const canvas: HTMLCanvasElement = <HTMLCanvasElement>document.getElementById("canvas");
+const res: CanvasRenderingContext2D | null = canvas.getContext("2d");
+if (res === null) {
+    throw Error("2d rendering context not supported");
+}
+const ctx: CanvasRenderingContext2D = res;
+
+let originPoint: Point;
+export let offSet: Point = new Point(0, 0); //No offset at the start
+let wasOut: boolean;
```
Comment (James-Oswald) [inline src/DragMode.ts:19]: comments on all global explaining what they are. We should be doing this in all mode files

File: src/DragMode.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+const canvas: HTMLCanvasElement = <HTMLCanvasElement>document.getElementById("canvas");
+const res: CanvasRenderingContext2D | null = canvas.getContext("2d");
+if (res === null) {
+    throw Error("2d rendering context not supported");
+}
+const ctx: CanvasRenderingContext2D = res;
+
+let originPoint: Point;
+export let offSet: Point = new Point(0, 0); //No offset at the start
```
Comment (James-Oswald) [inline src/DragMode.ts:18]: offset is one word

File: src/index.ts [hunk 3]
```diff
@@ -190,6 +201,11 @@ function mouseMoveHandler(event: MouseEvent) {
                 atomMouseMove(event);
             }
             break;
+        case "dragMode":
+            if (hasMouseDown && hasMouseIn) {
+                dragMouseMove(event);
+            }
+            break;
```
Comment (James-Oswald) [inline src/index.ts:208]: If we need to do this for all cases, we should move the if statement outside the switch statement.

File: src/index.html [hunk 4]
```diff
@@ -22,15 +22,15 @@
         <button type="button" title="Selection Mode" class="modeButton">
           <i class="fa fa-mouse-pointer" aria-hidden="true"></i>
         </button>
-        <button type="button" title="Move Mode" class="modeButton">
+        <button type="button" onclick= "dragMode()" title="Move Mode" class="modeButton">
           <i class="fa fa-arrows" aria-hidden="true"></i>
```
Comment (James-Oswald) [inline src/index.html:26]: lets change this icon to a hand maybe 
https://forkaweso.me/Fork-Awesome/icon/hand-paper-o/

**Review by James-Oswald — APPROVED** [review]

File: src/index.html (hunk 4 above)
Comment (DawnTheWitch) [reply src/index.html:26]: Oh I was wondering how this was done, very cool feature of HTML

**Pull request merged**